source/extensions/common/wasm/context.cc

```diff
@@ -1481,15 +1392,18 @@ void Context::log(const Http::HeaderMap* request_headers, const Http::HeaderMap*
   // ? request_trailers  ?
   access_log_response_headers_ = response_headers;
   access_log_response_trailers_ = response_trailers;
-  access_log_stream_info_ = &stream_info;
+  setStreamInfo(stream_info);
+  setRequestHeaders(request_headers);
+  setResponseHeaders(response_headers, response_trailers);
 
   onLog();
 
   access_log_request_headers_ = nullptr;
   // ? request_trailers  ?
   access_log_response_headers_ = nullptr;
   access_log_response_trailers_ = nullptr;
-  access_log_stream_info_ = nullptr;
+  clearRequestHeaders();
```

Comment:
Clear the stream info  as well.

Reply:
I am actually wondering if this is the right place. Is OnLog the last operation before the request/connection is discarded?